hawkbit-sdk/hawkbit-sdk-dmf/src/main/java/org/eclipse/hawkbit/sdk/dmf/amqp/Amqp.java

```diff
@@ -38,20 +39,35 @@ public void stop() {
         vHosts.values().forEach(VHost::stop);
     }
 
-    public VHost getVhost(final DMF dmf) {
+    public VHost getVhost(final DMF dmf, final boolean isEnvLocal) {
         final String vHost = dmf == null || ObjectUtils.isEmpty(dmf.getVirtualHost()) ?
                 (rabbitProperties.getVirtualHost() == null ? "/" :rabbitProperties.getVirtualHost()) :
                 dmf.getVirtualHost();
-        return vHosts.computeIfAbsent(vHost, vh -> {
-            final CachingConnectionFactory connectionFactory = new CachingConnectionFactory(rabbitProperties.getHost());
-            connectionFactory.setUsername(
-                    dmf == null || ObjectUtils.isEmpty(dmf.getUsername()) ?
-                            rabbitProperties.getUsername() : dmf.getUsername());
-            connectionFactory.setPassword(
-                    dmf == null || ObjectUtils.isEmpty(dmf.getPassword()) ?
-                            rabbitProperties.getPassword() : dmf.getPassword());
-            connectionFactory.setVirtualHost(vHost);
-            return new VHost(connectionFactory, amqpProperties);
-        });
+        return vHosts.computeIfAbsent(vHost, vh -> new VHost(getConnectionFactory(dmf), amqpProperties, isEnvLocal));
+    }
+
+    private ConnectionFactory getConnectionFactory(final DMF dmf) {
+        final String vHost = dmf == null || ObjectUtils.isEmpty(dmf.getVirtualHost()) ?
```

Comment:
Pass here String vHost already calculated in getVhost, or better, return it back in single method as before - so using vHost directly
Here vHost is resolved from DMF twice.

Reply:
passed string to getConnectionFactory.